### DISCUSSION TITLE
Webpack 5 & fs.readFile

### DISCUSSION THREAD

**Discussion by davidkaneda** [vercel/next.js#24327] [Help]
I'm not sure if this is more of a Next.js issue or Vercel, but our app breaks when adding `future.webpack5` into our Next.js config. There's only one issue, which is with sending mail, where we use fs.readFile to read our email template. Note that everything works fine in Webpack 4.

According to [the advanced usage section](https://vercel.com/docs/runtimes#advanced-usage/technical-details/including-additional-files) in Vercel's docs, static analysis is used to figure out which files to include with a build, so we were hoping this would just work after upgrading to Webpack 5. I also tried adding this to a vercel.json:

```
{
  "functions": {
    "pages/api/me/token.js": {
      "includeFiles": "public/email-templates/**"
    }
  }
}
```

But we get the same error:

```
2021-04-20T18:01:40.157Z	73994f3b-7d2f-4a2b-aacb-4f12c8a61cf0	ERROR	Error: ENOENT: no such file or directory, open '/var/task/public/email-templates/transactional.mjml'
```

Has anyone else has run into issues with `fs` after upgrading?

**Comment by martinbavio** [comment]
We have filed an issue for this, which has a more detailed description of the case, along with a repo to reproduce it: https://github.com/vercel/next.js/issues/25431
